infrastructure/resource_group_level/modules/cosmosdb.bicep

```diff
@@ -537,5 +538,21 @@ resource mongoDbPrivateEndpoint 'Microsoft.Network/privateEndpoints@2022-11-01'
   }
 }
 
+// Adding the Zone group sets up automatic DNS for the private link. 
+resource zoneGroup 'Microsoft.Network/privateEndpoints/privateDnsZoneGroups@2022-11-01' = {
```

Comment:
Note that using Zone Groups is the recommended way of setting up A records, but I think I might have to rewrite this to use the manual method that the CLI has because it isn't quite identical. Worth thinking about.